README.md

```diff
@@ -1,85 +1,341 @@
+        IterableAPI.register(token: deviceToken)
+    }
+    ```
+
+	Objective-C:
+	
+	```objective-c
+	- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+    	[IterableAPI registerToken:deviceToken];
+	}
+    ```
+    See example in sample app delegate [here](./sample-apps/swift-sample-app/swift-sample-app/AppDelegate.swift).
 
+Congratulations! You can now send remote push notifications to your device from Iterable!
+
+# Using the SDK
+
+1. ##### Handle Push Notifications
+
+	When the user taps on the push notification or one of the action buttons, the system calls `UNUserNotificationCenterDelegate`'s [userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:](https://developer.apple.com/documentation/usernotifications/unusernotificationcenterdelegate/1649501-usernotificationcenter?language=objc). Pass this call to **`IterableAppIntegration`** to track push open event and perform the associated action (see below for custom action and URL delegates).
+	
+	Swift:
+	
+	```swift
+	public func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+   		IterableAppIntegration.userNotificationCenter(center, didReceive: response, withCompletionHandler: completionHandler)
+   }
 	```
-	IterableAPI.email = 'user@example.com'
+
+	Objective-C:
+	
+	```objective-c
+	- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
+	    [IterableAppIntegration userNotificationCenter:center didReceiveNotificationResponse:response withCompletionHandler:completionHandler];
+	}
 	```
+	
+2. ##### Deep Linking
+	
+	* Handling Links from Push Notifications
+		
+		Push notifications and action buttons may have `openUrl` actions attached to them. When a URL is specified, the SDK first calls `urlDelegate` specified in your `IterableConfig` object. You can use this delegate to handle `openUrl` actions the same way as you handle normal deep links. If the delegate is not set or if it returns `false` (the default), the SDK will open Safari with that URL. If you want to navigate to a UIViewController on receiving a deep link, you should do so in the `urlDelegate`. 
+		
+		In the code below, `DeepLinkHandler` is a custom handler which is reponsible for deep link navigation. You have to provide implementation for deep link navigation. Please see [sample application](./sample-apps/swift-sample-app/swift-sample-app.xcworkspace) for a reference implementation.
+		
+		Swift:
+		
+		```swift
+		func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+			...
+			// Initialize Iterable API
+			let config = IterableConfig()
+			...
+			config.urlDelegate = self
```

Comment:
� before 'config' here. Also not sure if it's a Github preview bug or something..